src/iwant_bot/start.py

```diff
@@ -24,64 +51,336 @@ def format_message(message):
     text += ['We got these entries so far:\n']
     text += ['Or you can test POST request by command:']
     text += ['> curl -X POST  --data "user_name=0&channel_id=1&channel_name=2']
-    text += ['  &service_id=3&']
     text += ['  team_domain=4&team_id=5&text=6" http://localhost:8080']
     text.extend([format_message(msg) for msg in messages_we_got_so_far])
     return web.Response(text='\n'.join(text))
 
 
-def format_response(message='No response.'):
-    return json.dumps({'text': message})
+async def handle_button(request):
+    payload = body_to_dict(await request.post())
+    #print(payload)
+    body = json.loads(payload['payload'])
+    actions = body['actions']
+    print(actions)
 
+    if actions[0]['name'] == 'Invite' and actions[0]['type'] == 'button' and actions[0]['value'] != '0':
+        # upgrade and/or create request with previous `callback_id`
+        upgrade_iwant_request(body['callback_id'])
+        return web.json_response(body=format_response(f"Add request: {actions[0]['value']}"))
+    elif actions[0]['name'] == 'Invite' and actions[0]['type'] == 'button' and actions[0]['value'] == '0':
+        return web.json_response(body=format_response(f"Invitation was rejected."))
+    elif actions[0]['name'] == 'Time' and actions[0]['type'] == 'button':
+        for i in (body['callback_id']):
+            upgrade_iwant_request(body['callback_id'])
+            print(f"change duration of request to {actions[0]['value']}")
+        return web.json_response(body=format_response(
+            f"Duration of activity was set to {actions[0]['value']} seconds"))
+    else:
+        return web.json_response(body=format_response(f"Unknown action."))
```

Comment:
This is a good proof of concept, but the final code should have more layers (parse the request -> determine what to do -> do the concrete thing). Having multiple separate layers makes testing and customization possible.